course/viewbase.py

```diff
@@ -74,15 +76,20 @@ def get_resource_objects(self):
             # Try to find a language that is defined for this course instance
             # and apply it
             if self.instance.language:
-
+                instance_languages = self.instance.language.strip('|').split('|')
                 languages = []
+                url_lang = self.request.GET.get('hl')
+                if url_lang:
+                    if url_lang in instance_languages:
```

Comment:
This would benefit from the `instance_languages`, which is a set and not a list (i.e. you should have moved all 3 lines.